hw/ip/otbn/doc/dv/index.md

```diff
@@ -122,7 +122,16 @@ All four of these events should be crossed with the three states of the call sta
 #### Loop stack
 
 The [loop stack]({{< relref ".#loop-stack" >}}) is accessed by executing `LOOP` and `LOOPI` instructions.
-Important events for it are tracked at those instructions, rather than separately.
+Events concerning the start of loops are tracked at those instructions, but we can't track things like loop completion there.
+
+We expect to:
+- Complete a loop.
+- Complete the last loop on the stack (emptying it) after the stack has been full.
+- Complete a loop with one instruction and an iteration count of one.
+- Complete a loop with the maximal number of iterations.
+  Obviously, this isn't a reasonable thing to do for real in a test (there's a 32-bit iteration counter!).
+  The testbench will have to force a signal to skip past some iterations.
+- Run through a "badly nested" loop, where the body contains the final instruction from an outer loop (checking that we don't wrongly skip back).
```

Comment:
Maybe we want some stuff around branching/jumps too?

- Branch in a loop
- Branch out of immediate loop body (i.e. out of instructions with addresses between start and end loop address) to return later
- Function call/jump in immediate loop body to return later

May be no point differentiating the second two as they're both just control flow changes, the details of what opcode triggered them and whether we're using the call stack shouldn't matter.

Do we also want an explicit 'See well-behaved loop nesting'? Other cover points require loop nesting to hit but are all around extreme behaviours. I guess 'Complete the last loop on the stack (emptying it) after the stack has been full' already gives us well-behaved loop nesting.

Reply:
Good points. The "Branch in a loop" check is already tracked (on the branch instruction), but we don't have the others yet. I've also just realised we should probably catch "jump to final address in a loop", which feels like a corner case. I'll push an update with these, probably tomorrow.